src/main/java/org/opentripplanner/middleware/models/TripHistoryUpload.java

```diff
@@ -0,0 +1,66 @@
+package org.opentripplanner.middleware.models;
+
+import com.mongodb.client.FindIterable;
+import com.mongodb.client.model.Filters;
+import com.mongodb.client.model.Sorts;
+import org.bson.codecs.pojo.annotations.BsonIgnore;
+import org.bson.conversions.Bson;
+import org.opentripplanner.middleware.cdp.TripHistoryUploadStatus;
+import org.opentripplanner.middleware.persistence.Persistence;
+
+import java.util.Date;
+
+public class TripHistoryUpload extends Model {
+
+    public Date uploadDate;
+    public String status = TripHistoryUploadStatus.PENDING.getValue();
+
+    /** This no-arg constructor exists to make MongoDB happy. */
+    public TripHistoryUpload() {
+    }
+
+    public TripHistoryUpload(Date uploadDate) {
+        this.uploadDate = uploadDate;
+        this.status = TripHistoryUploadStatus.PENDING.getValue();
+    }
+
+    /**
+     * Get all incomplete uploads.
+     */
+    @BsonIgnore
+    public static FindIterable<TripHistoryUpload> getIncompleteUploads() {
```

Comment:
It'd also be my personal preference to move these static methods into the classes where they're called from.